src/bit-systems/video-system.ts

```diff
@@ -1,126 +1,69 @@
-      swapAudioType(world, audioSystem, eid, StereoAudio);
-    }
-
-    applySettings(audio, settings);
-    removeComponent(world, AudioSettingsChanged, eid);
-  });
-}
+const video2audio = new Map<number, number>();
 
 const OUT_OF_SYNC_SEC = 5;
 const networkedVideoQuery = defineQuery([NetworkedVideo]);
 const mediaVideoQuery = defineQuery([MediaVideo]);
 const mediaVideoEnterQuery = enterQuery(mediaVideoQuery);
+const mediaVideoExitQuery = exitQuery(mediaVideoQuery);
+const mediaPlaybackUpdated = defineQuery([MediaVideo, MediaVideoPlaybackChanged]);
+const mediaLoadStatusQuery = defineQuery([MediaVideo, MediaLoaded]);
+const mediaLoadedQuery = enterQuery(mediaLoadStatusQuery);
+const mediaUnloadedQuery = exitQuery(mediaLoadStatusQuery);
 export function videoSystem(world: HubsWorld, audioSystem: AudioSystem) {
-  mediaVideoEnterQuery(world).forEach(function (eid) {
-    const videoObj = world.eid2obj.get(eid) as Mesh;
+  mediaVideoEnterQuery(world).forEach(function (videoEid) {
+    const videoObj = world.eid2obj.get(videoEid) as Mesh;
     const video = (videoObj.material as MeshStandardMaterial).map!.image as HTMLVideoElement;
-    if (MediaVideo.autoPlay[eid]) {
+    if (MediaVideo.autoPlay[videoEid]) {
       video.play().catch(() => {
         // Need to deal with the fact play() may fail if user has not interacted with browser yet.
         console.error("Error auto-playing video.");
       });
     }
-    const audio = world.eid2obj.get(makeAudioSourceEntity(world, video, audioSystem))!;
+    const audioEid = makeAudioSourceEntity(world, video, audioSystem);
+    video2audio.set(videoEid, audioEid);
+    const audio = world.eid2obj.get(audioEid)!;
     videoObj.add(audio);
     // Note in media-video we call updateMatrixWorld here to force PositionalAudio's updateMatrixWorld to run even
     // if it has an invisible parent. We don't want to have invisible parents now.
   });
-
-  audioEmitterSystem(world, audioSystem);
-
+  mediaLoadedQuery(world).forEach(videoEid => {
+    const audioEid = video2audio.get(videoEid)!;
+    const rootEid = MediaLoaded.rootRef[videoEid];
+    if (APP.audioOverrides.has(rootEid)) {
+      const audioSettings = APP.audioOverrides.get(rootEid)!;
+      APP.audioOverrides.set(audioEid, audioSettings);
+    }
+  });
+  mediaUnloadedQuery(world).forEach(videoEid => {
+    const audioEid = video2audio.get(videoEid)!;
+    APP.audioOverrides.delete(audioEid);
+  });
+  mediaVideoExitQuery(world).forEach(videoEid => {
+    const rootEid = MediaLoaded.rootRef[videoEid];
+    APP.audioOverrides.delete(rootEid);
+    video2audio.delete(videoEid);
+  });
+  mediaPlaybackUpdated(world).forEach(videoEid => {
+    const audioEid = video2audio.get(videoEid)!;
+    const videoEl = (world.eid2obj.get(videoEid) as any).material.map.image as HTMLVideoElement;
+    if (videoEl.paused) {
+      APP.isAudioPaused.add(audioEid);
+    } else {
+      APP.isAudioPaused.delete(audioEid);
```

Comment:
Whenever we change any of the audio settings maps on APP, we have to call updateAudioSettings, or else the changes won't be flushed to the audios.

Reply:
`isAudioPaused` doesn't affect the audio node settings so `updateAudioSettings` it's not necessary but this is confusing because and it might not be easy to know when calling `updateAudioSettings` it's required. This raises a good point. We have a few state stores (`isAudioPaused`, `clippingState`, `mutedState`, etc)  that we might be able to replace for flags in the `AudioEmitter` component. That would move that state to the right place and remove a bit of complexity on the maps. So I'm going to do that for `isAudioPaused` and `mutedState` which are the ones that we are touching now and move the rest as I migrate other parts.